src/generators/typescript/presets/CommonPreset.ts

```diff
@@ -0,0 +1,145 @@
+    const camelCasedProp = FormatHelpers.toCamelCase(prop);
+    const propUnmarshal = propModel.$ref !== undefined ? `${renderer.nameType(propModel.$ref)}.unmarshal(obj.${camelCasedProp})` : `obj.${camelCasedProp}`;
+    return `if (obj.${camelCasedProp} !== undefined) {
+  instance.${camelCasedProp} = ${propUnmarshal};
+}`;
+  }).join('\n');
+
+  let unmarshalPatternProperties = '';
+  if (model.patternProperties !== undefined) {
+    for (const [pattern, patternModel] of Object.entries(model.patternProperties)) {
+      let patternPropertyName = getUniquePropertyName(model, `${pattern}${DefaultPropertyNames.patternProperties}`);
+      patternPropertyName = renderer.nameProperty(patternPropertyName, patternModel);
+      unmarshalPatternProperties += `if (key.match(new RegExp('${pattern}'))) {
+  if (instance.${patternPropertyName} === undefined) {instance.${patternPropertyName} = new Map();}
+  instance.${patternPropertyName}.set(key, value as any);
+  continue;
+}`;
+    }
+  }  
+
+  let unmarshalAdditionalProperties = '';
+  if (model.additionalProperties !== undefined) {
+    let additionalPropertyName = getUniquePropertyName(model, DefaultPropertyNames.additionalProperties);
+    additionalPropertyName = renderer.nameProperty(additionalPropertyName, model.additionalProperties);
+    const additionalPropertiesCast = model.additionalProperties.$ref !== undefined ? `${renderer.nameType(model.$id)}.unmarshal(value)` : 'value as any';
+    unmarshalAdditionalProperties = `if (instance.${additionalPropertyName} === undefined) {instance.${additionalPropertyName} = new Map();}
+instance.${additionalPropertyName}.set(key, ${additionalPropertiesCast});`;
+  }
+  const formattedModelName = renderer.nameType(model.$id);
+  const propertyNames = Object.keys(properties).map((prop => `"${prop}"`));
+  return `public static unmarshal(json: string | object): ${formattedModelName} {
+  const obj = typeof json === "object" ? json : JSON.parse(json);
+  const instance = new ${formattedModelName}({});
+
+${renderer.indent(unmarshalProperties)}
+
+  //Not part of core properties
+  for (const [key, value] of Object.entries(obj).filter((([key,]) => {return ![${propertyNames}].includes(key);}))) {
+    //Check all pattern properties
+${renderer.indent(unmarshalPatternProperties, 4)}
+
+${renderer.indent(unmarshalAdditionalProperties, 4)}
+  }
+  return instance;
+}`;
+} 
+
+/**
+ * Preset which adds `marshal`, `unmarshal` functions to class. 
+ * 
+ * @implements {TypeScriptPreset}
+ */
+export const TS_COMMON_PRESET: TypeScriptPreset = {
+  class: {
+    additionalContent({ renderer, model, content, options }) {
+      options = options || {};
+      const blocks: string[] = [];
+      
+      if (options.marshal === undefined || options.marshal === true) {blocks.push(renderMarshal({ renderer, model }));}
+      if (options.unmarshal === undefined || options.unmarshal === true) {blocks.push(renderUnmarshal({ renderer, model }));}
```

Comment:
Should the two functions be disabled by default?

Reply:
This function should be disabled by default and I think that `marshal` and `unmarshal` should be applied together, not with different options, so `options.marshalling = true` should render these two functions.

Reply:
I think that makes sense yea 👍 Changing